modular_skyrat/modules/mold/code/mold.dm

```diff
@@ -0,0 +1,282 @@
+ * * amount_of_reagent - how much of the reagent to add to the reagents datum. Defaults to 30.
+ */
+/datum/mold/proc/spew_foam(obj/structure/mold/structure/source, range, reagent_capacity, reagent_to_add, amount_of_reagent = 30)
+	source.visible_message(span_warning("[source] spews out foam!"))
+	var/datum/reagents/spewed_reagents = new /datum/reagents(reagent_capacity)
+	spewed_reagents.my_atom = source
+	spewed_reagents.add_reagent(reagent_to_add, amount_of_reagent)
+	var/datum/effect_system/fluid_spread/foam/foam = new
+	var/turf/source_turf = get_turf(source)
+	foam.set_up(range, location = source_turf, carry = spewed_reagents)
+	foam.start()
+
+
+/**
+ * Fire mold
+ *
+ * A mold type that is based around a fire theme. It's the colour of fire, prefers hotter
+ * temperatures, spawns a burn mix from its conditioners. The whole shebang. The mobs it
+ * spawns are oil shamblers, a humanoid figure that can set fire to people on hit.
+ * Probably the most simple mold type.
+ */
+/datum/mold/fire
+	name = "fire"
+	mold_color = "#e04000"
+	structure_light_color = LIGHT_COLOR_FIRE
+	examine_text = "It feels hot to the touch."
+	mob_types = list(/mob/living/simple_animal/hostile/biohazard_blob/oil_shambler)
+	preferred_atmos_conditions = "co2=30;TEMP=1000"
+	resistance_flags = FIRE_PROOF
+
+/datum/mold/fire/core_defense(obj/structure/mold/structure/core/core)
+	core.visible_message(span_warning("[core] puffs out a cloud of flames!"))
+	spawn_atmos(core)
+
+/datum/mold/fire/bulb_discharge(obj/structure/mold/structure/bulb/bulb)
+	. = ..()
+	spawn_atmos(bulb)
+
+/datum/mold/fire/proc/spawn_atmos(obj/structure/mold/structure/source)
+	var/turf/source_turf = get_turf(source)
+	source_turf.atmos_spawn_air("o2=20;plasma=20;TEMP=600")
+
+
+/**
+ * Disease mold
+ *
+ * A mold type that centers around disease. It spawns rats that inject with disease, lets
+ * out puffs of fungal smoke that spread disease.
+ * It has its own custom disease, too. Fancy.
+ */
+/datum/mold/disease
+	name = "fungal"
+	tier = MOLD_TIER_HIGH_THREAT
+	mold_color = "#6e5100"
+	structure_light_color = LIGHT_COLOR_BROWN
+	examine_text = "It looks like it's rotting."
+	mob_types = list(/mob/living/simple_animal/hostile/biohazard_blob/diseased_rat)
+	spawn_cooldown = 5 SECONDS
+	preferred_atmos_conditions = "TEMP=312"
+
+/datum/mold/disease/core_defense(obj/structure/mold/structure/core/core)
+	core.visible_message(span_warning("[core] emits a cloud!"))
+	fungal_puff(core, 5)
+
+/datum/mold/disease/bulb_discharge(obj/structure/mold/structure/bulb/bulb)
+	. = ..()
+	fungal_puff(bulb, 4)
+
+/datum/mold/disease/proc/fungal_puff(source, range)
+	var/datum/reagents/reagents = new/datum/reagents(300)
+	reagents.my_atom = source
+	reagents.add_reagent(/datum/reagent/cryptococcus_spores, 50)
+	var/datum/effect_system/fluid_spread/smoke/chem/smoke_machine/puff = new
+	var/turf/source_turf = get_turf(source)
+	puff.set_up(range, location = source_turf, carry = reagents, efficiency = 24)
```

Comment:
Same deal with these three.

I'm not sure why 300 was used here to begin with, if it's not needed I guess we can lower it, or we can just make a define for a default temporary reagents holder.